src/Dict/Any.elm

```diff
@@ -631,3 +631,146 @@ encodeList encodeF =
 decodeList : (k -> comparable) -> Decode.Decoder ( k, v ) -> Decode.Decoder (AnyDict comparable k v)
 decodeList keyToComparable =
     Decode.map (fromList keyToComparable) << Decode.list
+
+
+{-| Takes a key-fn and a list.
+Creates an `AnyDict` which maps the key to a list of matching elements.
+-}
+groupBy : (value -> key) -> (key -> comparable) -> List value -> AnyDict comparable key (List value)
+groupBy toKey keyToComparable list =
+    List.foldr
+        (\x acc ->
+            update
+                (toKey x)
+                (\maybeValues ->
+                    maybeValues
+                        |> Maybe.map ((::) x)
+                        |> Maybe.withDefault [ x ]
+                        |> Just
+                )
+                acc
+        )
+        (empty keyToComparable)
+        list
+
+
+{-| Find out if there is any instance of something in a Dictionary.
+
+    type Animal = Cat | Mouse | Dog
+
+    animalToInt : Animal -> Int
+    animalToInt animal =
+        case animal of
+            Cat -> 0
+            Mouse -> 1
+            Dog -> 2
+
+    animals : AnyDict Int Animal String
+    animals =
+        [ (Cat, "Tom"), (Mouse, "Jerry") ]
+            |> fromList animalToInt
+
+    isACat : Animal -> String -> Bool
+    isACat animal _ =
+        case animal of
+            Cat -> True
+            _ -> False
+
+    any isACat animals
+    --> True
+
+-}
+any : (k -> v -> Bool) -> AnyDict comparable k v -> Bool
+any predicate dict =
+    foldl
+        (\k v acc -> acc || predicate k v)
+        False
+        dict
+
+
+{-| Find out if all instances of a Dictionary match a predicate.
+
+    type Animal = Cat | Mouse | Dog
+
+    animalToInt : Animal -> Int
+    animalToInt animal =
+        case animal of
+            Cat -> 0
+            Mouse -> 1
+            Dog -> 2
+
+    animals : AnyDict Int Animal String
+    animals =
+        [ (Cat, "Tom"), (Mouse, "Jerry") ]
+            |> fromList animalToInt
+
+    aristocats : AnyDict Int Animal String
+    aristocats =
+        [ (Cat, "Marie"), (Cat, "Duchess"), (Cat, "Toulouse"), (Cat, "Berlioz") ]
+            |> fromList animalToInt
+
+    isACat : Animal -> String -> Bool
+    isACat animal _ =
+        case animal of
+            Cat -> True
+            _ -> False
+
+    all isACat animals
+    --> False
+
+    all isACat aristocats
+    --> True
+
+-}
+all : (k -> v -> Bool) -> AnyDict comparable k v -> Bool
+all predicate dict =
+    foldl
+        (\k v acc -> acc && predicate k v)
+        True
+        dict
```

Comment:
if I'm correct this function doesn't exist for Dict even in dict-extra, right?

Reply:
Right, for some reason they only implemented “any” 🤔